src/Windows/Avalonia.Win32/NonPumpingSyncContext.cs

```diff
@@ -32,23 +31,15 @@ public override int Wait(IntPtr[] waitHandles, bool waitAll, int millisecondsTim
 
         public void Dispose() => SetSynchronizationContext(_inner);
 
-        public static IDisposable Use()
+        public static IDisposable? Use()
         {
             var current = Current;
-            if (current == null)
```

Comment:
This sync context is required to prevent various side effects of Monitor.Enter and other APIs that occur only on STA threads on Windows. 

It's required regardless of another sync context existing on the thread previously, if the current thread is an STA thread, we have to install our own that uses non-alertable waits.

Reply:
In the previous code, if `current is null` and `state == STA`, you get to `new NonPumpingSyncContext(current)` which is `NonPumpingSyncContext(null)`. That sync context will crash with an NRE on the first `Post`/`Send` call, hence the changes.

If I'm understanding correctly, you're telling me that `Post`/`Send` won't be called (indeed in the current scoped usages they wont): this sync context is only used to set the `IsWaitNotificationRequired` flag. In that case, let me change that `_inner` to a nullable field, and use `inner?.Post/Send` instead.

Reply:
OK, we need to fix `NonPumpingSyncContext` to handle `null` then

Reply:
`Post` can be implemented as `ThreadPool.QueueUserWorkItem`, `Send` can invoke the callback immediately. IIRC that's the default behavior when there is no synchronization context.

Reply:
The STA check has been restored, and `Send/Post` implemented as mentioned.